docs/http-status-codes.adoc

```diff
@@ -0,0 +1,166 @@
+:showtitle:
+:numbered:
+:toc: left
+
+= HTTP Status Codes used in Omicron
+
+This doc is aimed at the public API.  For consistency, we should use the same error codes for internal APIs as well.
+
+== Summary
+
+* "200 OK" is used for most successful GET and PUT requests.
+* "201 Created" is used for successful POST requests that create resources.
+* "202 Accepted" is used for successful POST requests that do not create resources (e.g., attaching a disk).
+* "204 No Content" is used for most successful DELETE requests.
+* "400 Bad Request" is used for most sorts of input validation error.
+* Regarding access control:
+** "401 Unauthorized" is used when the user provided no credentials or bad credentials for an operation that obviously requires authentication (which is most of them).  It's _not_ used when valid credentials are provided.  That is, this reflects an _authentication_ problem.
```

Comment:
(no action) this is the worst

Reply:
Agreed!  I gather it [maybe predates](https://security.stackexchange.com/questions/134113/http-authentication-the-http-header-name-is-authorization-why) the industry's (re)definition of authentication and authorization.